components/amount-buttons.jsx

```diff
@@ -53,7 +60,36 @@ var AmountOtherButton = React.createClass({
       this.props.onRadioChange();
     }
   },
+  onInputChange: function(e) {
+    var inputValue = e.currentTarget.value;
+    var amount = "";
+
+    if (/^(\d)*[\,]?(\d){0,2}$/.test(inputValue)) {
```

Comment:
Can you add a comment above this regex that you are trying to capture as many number follow by comma and follow by 0 to 2 numbers?